compiler/qsc_rca/src/lib.rs

```diff
@@ -659,7 +661,7 @@ impl ValueKind {
                 }
                 ValueKind::Element(self_runtime_kind) => {
                     *content_runtime_kind = self_runtime_kind;
-                    *size_runtime_kind = self_runtime_kind;
+                    *size_runtime_kind = RuntimeKind::Static;
```

Comment:
```suggestion
                    // When we project from an element variant to an array variant, we assume the size of the
                    // array is static because we rely on the dynamically-sized arrays runtime feature flag to
                    // detect such cases.
                    *size_runtime_kind = RuntimeKind::Static;
```